js/kraken.js

```diff
@@ -815,22 +809,11 @@ module.exports = class kraken extends Exchange {
         const currencyIds = Object.keys (balances);
         for (let i = 0; i < currencyIds.length; i++) {
             const currencyId = currencyIds[i];
-            let code = currencyId;
```

Comment:
We can't really drop this section for Kraken, because we can't hardcode all of their currencies. I would suggest to keep the logic that cuts off the first letter for kraken, but, maybe, detach it into a separate method... What do you think?

Reply:
nope, i think we can hardcode all their currencies, since they do not have very many pairs (around 8). it is also much speedier than doing string operations.

Reply:
moreover not all the quoteIds have to strange 'X' / 'Z' prefix, only some do. For example EUR is 'EUR' where 'ZGBP' is 'GBP'

Reply:
> since they do not have very many pairs (around 8)

I'm talking of base currencies as well, not just quote currencies. I think you're missing LTC, GNO, MLN, QTUM and + 50% of all other currencies. Maintaining a hardcoded list of those is not what we want, because they will list and delist them arbitrarily.

Reply:
One of possible solutions – we could override `safeCurrencyCode` for Kraken specifically... ?

Reply:
it is not necessary, most currencies for kraken are standard and do not have the 'X' prefix - try for yourself.

Reply:
Well, you have missed a few... and we will always be behind Kraken if we do this...

```
mbp:ccxt igorkroitor$ node examples/js/cli kraken currencies --table
  id | code | name | active | fee | precision |                                                                                            limits
-------------------------------------------------------------------------------------------------------------------------------------------------
...
XDAO |  DAO |  DAO |   true |     |        10 |                                                                                   [object Object]
XICN |  ICN |  ICN |   true |     |        10 |                                                                                   [object Object]
XLTC |  LTC |  LTC |   true |     |        10 |                                                                                   [object Object]
XMLN |  MLN |  MLN |   true |     |        10 |                                                                                   [object Object]
XNMC |  NMC |  NMC |   true |     |        10 |                                                                                   [object Object]
XXDG | DOGE | DOGE |   true |     |         8 |                                                                                   [object Object]
XXLM |  XLM |  XLM |   true |     |         8 |                                                                                   [object Object]
XXMR |  XMR |  XMR |   true |     |        10 |                                                                                   [object Object]
XXRP |  XRP |  XRP |   true |     |         8 |                                                                                   [object Object]
XXVN |  XVN |  XVN |   true |     |         4 |                                                                                   [object Object]
XZEC |  ZEC |  ZEC |   true |     |        10 |                                                                                   [object Object]
ZCAD |  CAD |  CAD |  false |     |         4 |                                                                                   [object Object]
ZJPY |  JPY |  JPY |  false |     |         2 | {"amount":{"min":0.01,"max":100},"price":{"min":0.01,"max":100},"cost":{},"withdraw":{"max":100}}
ZKRW |  KRW |  KRW |   true |     |         2 | {"amount":{"min":0.01,"max":100},"price":{"min":0.01,"max":100},"cost":{},"withdraw":{"max":100}}
```

I suggest that we don't hardcode the entire list, because that's more than 50% of all their currencies... 

I still think we should override safeCurrencyCode for Kraken:

```JavaScript
safeCurrencyCode (currencyId, currency = undefined) {
    if ((currencyId.length > 3) && (currencyId.indexOf ('X') === 0)) {
        currencyId = currencyId.slice (1);
    }
    return super.safeCurrencyCode (currencyId, currency);
}
```

What do you say?

Reply:
sure, will add this now